sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -259,11 +256,151 @@ class AzureAppConfigurationProvider(Mapping[str, Union[str, JSON]]):
+        self._selects: List[SettingSelector] = kwargs.pop(
+            "selects", [SettingSelector(key_filter="*", label_filter=EMPTY_LABEL)]
+        )
+        self._sentinel_keys: List[str] = []
+
+        if self._refresh_options is not None:
+            self._next_refresh_time = datetime.now() + timedelta(
+                seconds=self._refresh_options._refresh_interval
+            )
+
+        trim_prefixes: List[str] = kwargs.pop("trim_prefixes", [])
+        self._trim_prefixes = sorted(trim_prefixes, key=len, reverse=True)
+        self._attempts = 1
+        self._min_backoff: Optional[int] = kwargs.pop("min_backoff", 30)
+        self._max_backoff: Optional[int] = kwargs.pop("min_backoff", 600)
+
+    def refresh(self) -> None:
+        if (
+            self._refresh_options is None
+            or len(self._refresh_options.get_refresh_registrations()) == 0
+        ):
+            return
+
+        if datetime.now() < self._next_refresh_time:
+            return
+
+        refresh_all = False
+
+        for registration in self._refresh_options.get_refresh_registrations():
+            if registration.refresh_all == True:
+                updated_sentinel = self._client.get_configuration_setting(
+                    key=registration.key_filter, label=registration.label_filter
+                )
+                if registration.etag == updated_sentinel.etag:
+                    continue
+                else:
+                    refresh_all = True
+                    break
+
+        if refresh_all:
+            self._load_all()
+            self._next_refresh_time = datetime.now() + timedelta(
+                seconds=self._refresh_options._refresh_interval
+            )
+            return
+
+        # Only update individual keys if the refresh_all didn't trigger
+        updated_dict = self._dict.copy()
+        updated_registrations = self._refresh_options.get_refresh_registrations().copy()
+        
+        try:
+            for registration in updated_registrations:
+                if registration.refresh_all == False:
+                    updated_sentinel = self._client.get_configuration_setting(
+                        key=registration.key_filter, label=registration.label_filter
+                    )
+                    if registration.etag == updated_sentinel.etag:
+                        continue
+                    else:
+                        registration.etag = updated_sentinel.etag
+                        updated_dict[
+                            self._proccess_key_name(updated_sentinel)
+                        ] = self._proccess_key_value(updated_sentinel)
+            
+            self._dict = updated_dict
+            self._refresh_options.set_refresh_registrations(updated_registrations)
+            self._next_refresh_time = datetime.now() + timedelta(
+                seconds=self._refresh_options._refresh_interval
+            )
+            self._attempts = 1
+        except:
```

Comment:
At a minimum, this should be caught and logged rather than silently ignored.
However why would we not just raise here? Wouldn't I want to know immediately that my config is failing to refresh?
Additionally, a naked except like this makes no distinction for a recoverable vs unrecoverable error.

Reply:
For the logging, just hadn't gotten around to adding it. Only meant this to show what the idea was.

We don't raise an exception here for a number of reasons. Mainly there is nothing that the customer should do with such an exception. We don't want them to retry the operation, yet. They should just call it whenever. When refresh interval is updated here it can be anywhere from the default of 30 seconds to 10 minutes before we will even try again.

In addition, the chance of this failing will be reduced greatly once we add geo-replication, which is on our backlog.

Reply:
When you say "there is nothing that the customer should do with such an exception" - then if we've reached this point in the application, any error encountered _must_ be transitory and _must_ be retyable?
I.e. there's no possibility of an unrecoverable error?
Typically, in scenarios where we don't surface errors directly (e.g. where we might be processing events, or batching requests in the background etc), we still offer some kind of `on_error` callable parameter to give customers some flexibility here. Some of this also ties into how we present retry options, an alternative might be having some kind of "max_reties=0" for a "raise on any error" scenario.